applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -135,38 +144,94 @@ static void handle_dvfs_error(int32_t err)
 	} else if (err == -EAGAIN) {
 		LOG_DBG("DVFS not initialized, trying again.");
 		timeout = DVFS_RETRY_INITIALIZATION_TIMEOUT;
+	} else if (err == -ETIMEDOUT) {
+		LOG_DBG("DVFS frequency change timed out, trying again.");
+		timeout = DVFS_RETRY_BUSY_TIMEOUT;
```

Comment:
Do we know when could this happen? I wonder if we should retry or make it a fatal error.

Reply:
After looking a bit more into implementation, I started to wonder if fatal error would not be a better choice here (sorry for missing it initially). I noticed that `-ETIMEDOUT` is triggered on a timeout that is internal to `nrf_clock` driver (when there is no response from dvfs). I am not sure if retrying in that case would be better choice. @zycz, @mkapala-nordic what's your opinion?

Reply:
IMO if there is some lag on communication beetwen app <-> sysctrl or driver for some other case do not meet this timeout sth is very wrong and we should go to error state. If sysctrl is busy or dvfs is not initialized then we have other errors. But I am not strongly convinced to that so for me both options are acceptable.

Reply:
Then maybe we should use error state here? (if needed we can always configure the timeout value in driver's configuration)

Reply:
@mkapala-nordic you also preferred retrying - what do you think? Should I leave it as it is (retrying) or revert it back to original solution (fatal error)

Reply:
won't we enter into the fatal error anyway if we retry too much?

Reply:
Ok so lets say that @MarekPieta and me do not have a strong opinion here and you have so lets leave it as it is (retrying).

Reply:
I am worried that we might enter a strange state by retrying multiple times on timeout (the timeout would suggest that  something is wrong with the system controller or communication - in general we should get a response)

Reply:
@MarekPieta @mkapala-nordic  please talk it over which option you prefer.

Reply:
I asked @anangl. He also suggest to treat the timeout as a fatal error as it's rather impossible that subsequent try would succeed.

Reply:
Ok, will return back to my first solution then.